DocumentsFromSnapshotMigration/src/test/java/org/opensearch/migrations/bulkload/CustomTransformationTest.java

```diff
@@ -0,0 +1,282 @@
+                    .withNetwork(network)
+                    .withNetworkAliases(TARGET_DOCKER_HOSTNAME);
+        ) {
+            CompletableFuture.allOf(
+                CompletableFuture.runAsync(esSourceContainer::start),
+                CompletableFuture.runAsync(osTargetContainer::start)
+            ).join();
+
+            var sourceClusterOperations = new ClusterOperations(esSourceContainer.getUrl());
+
+            var shards = 2;
+            // Number of default shards is different across different versions on ES/OS.
+            // So we explicitly set it.
+            String body = String.format(
+                    "{" +
+                            "  \"settings\": {" +
+                            "    \"index\": {" +
+                            "      \"number_of_shards\": %d," +
+                            "      \"number_of_replicas\": 0" +
+                            "    }" +
+                            "  }" +
+                            "}",
+                    shards
+            );
+            sourceClusterOperations.createIndex("geonames", body);
+            sourceClusterOperations.createDocument("geonames", "111", "{\"author\":\"Tobias Funke\", \"category\": \"cooking\"}");
+
+            // Create the snapshot from the source cluster
+            var args = new CreateSnapshot.Args();
+            args.snapshotName = SNAPSHOT_NAME;
+            args.fileSystemRepoPath = SearchClusterContainer.CLUSTER_SNAPSHOT_DIR;
+            args.sourceArgs.host = esSourceContainer.getUrl();
+
+            var snapshotCreator = new CreateSnapshot(args, testSnapshotContext.createSnapshotCreateContext());
+            snapshotCreator.run();
+
+            esSourceContainer.copySnapshotData(tempDirSnapshot.toString());
+
+            int exitCode;
+            int finalExitCodeCount = 0;
+            int runs = 0;
+            do {
+                exitCode = processRunner.apply(new RunData(tempDirSnapshot, tempDirLucene, osTargetContainer));
+                runs++;
+                if (exitCode == eventualExitCode) {
+                    finalExitCodeCount++;
+                }
+                log.atInfo().setMessage("Process exited with code: {}").addArgument(exitCode).log();
+                // Clean tree for subsequent run
+                deleteTree(tempDirLucene);
+            } while (finalExitCodeCount < eventualExitCodeCount && runs < initialExitCodes * 2);
+
+            // Assert doc count on the source and target cluster match expected
+            validateFinalClusterDocs(
+                esSourceContainer,
+                osTargetContainer,
+                DocumentMigrationTestContext.factory().noOtelTracking(),
+                expectedSourceDocs,
+                expectedTargetDocs
+            );
+        } finally {
+            deleteTree(tempDirSnapshot);
+        }
+    }
+
+    private static String createIndexNameTransformation(String existingIndexName, String newIndexName) {
```

Comment:
Can you please add a comment explaining what this transform is supposed to accomplish?  The method name is a good start, but, frankly, Jolt is inscrutable.

Reply:
Added